lib/trino-parquet/src/main/java/io/trino/parquet/ParquetTypeUtils.java

```diff
@@ -291,6 +294,13 @@ private static Optional<Field> constructField(Type type, ColumnIO columnIO, bool
         boolean required = columnIO.getType().getRepetition() != OPTIONAL;
         int repetitionLevel = columnIO.getRepetitionLevel();
         int definitionLevel = columnIO.getDefinitionLevel();
+        if (isVariantType(type, columnIO)) {
+            checkArgument(type.getTypeParameters().isEmpty(), "Variant type should not have parameters");
+            GroupColumnIO groupColumnIo = (GroupColumnIO) columnIO;
```

Comment:
if (!(columnIo instanceof GroupColumnIO groupColumnInfo)) {
    throw new IllegalStateException("Expected columnIO to be GroupColumnIO but got %s".formatted(columnIo.getClass().getSimpleName());
}